### ISSUE TITLE
Combo layer activated by two keys but held using one

### ISSUE THREAD

**Issue by lobre** [rvaiya/keyd#703]
The more I experiment about keyboards and layers, the more I am thinking about particular gestures that could be worth trying to enable new workflows using two keys.

I have had a new idea that I would like to try but I am not sure it is currently doable in keyd, hence this issue to try to figure it out, and potentially see what would be missing.

"Combos" usually mean pressing two keys simultaneously to generate a behavior. In keyd, they can be described using `a+b = ...` syntax.

Some people don't like combos because pressing or holding two keys can already be a overhead, and I share this feeling.

So I am wondering if a gesture such as following could be doable, to simplify a little bit the combo. Note that it applies only to combos that are meant to be held, such as when activating a layer.

For the example, let's presume we have two keys named `a` and `b` assigned to the two thumbs. `a` can be tapped using left thumb, and `b` using right thumb. 

- Press key `a` and `b` simultaneously. This enables my target custom layer while they are held.
- Now this custom layer will stay enabled if I keep either `a` or `b` held, releasing the other.

This way, the layer is activated using the combo, but they I can release one of the two thumb keys and keep only one pressed, which simplifies the process a little bit because it frees one hand.

To sum up, the combo would be used just for the activation, but then the layer would be held using only one of the two keys of the combo (such as when using a regular held `layer(...)`).

The advantage is that taking two thumb keys, I can activate up to 3 different layers, always ending with a "single thumb hold".

- Hold only key `a` would gives `layer1`.
- Hold only key `b` would gives `layer2`.
- Press `a` and `b` simultaneously and release one or the other would give `layer3`.

What do you think of this idea? And do you think I can configure keyd in a way that mimics this behaviour?

**Comment by lobre** [comment]
My bad. I was away from the computer when submitting this issue and I did not try implementing it.

I realize today that what I what is actually **the default behavior of chords in keyd**...

To release the chord, one has to let go all keys from the chord. If only one is released, the chord will stay active. See the code:

https://github.com/rvaiya/keyd/blob/8be57cdf22adec6843a091ec705c5906feeadc68/src/keyboard.c#L916

So I will close this issue as there is nothing to do to have this. Sorry about the noise.
